### ISSUE TITLE
Unexpected escape of `fetch.ubuntu._run_with_retries` without successfully finishing the command

### ISSUE THREAD

**Issue by addyess** [juju/charm-helpers#743]
[`fetch.ubuntu._run_with_retries`](https://github.com/juju/charm-helpers/blob/f086257853889220a042aa451e528eb18e10cf7a/charmhelpers/fetch/ubuntu.py#L907) seems to escape raising a failure in the event that the `subprocess.check_call` yields an exit code which is a failed exit code, but not in the list of `retry_exitcode`

this gives this method a false success when in reality it should fail immediately (no retrying)

i think the library method should raise when the exit value of `subprocess.check_call` is an error not on the rety list

TLDR:
if its exit value is 0 -- :tada:
if the exit value is in the `retry_results` -- retry max_retries like expected
if the exit value is not in the `retry_results` -- :fire:
